src/fmu/tools/rms/copy_rms_param_to_ertbox_grid.py

```diff
@@ -1517,3 +1531,360 @@ def set_continuous_3d_parameter_values_in_zone_region(
+    }
+
+
+def get_grid_building_job_arguments(
+    grid_model_name: str, rms_grid_job_name: str
+) -> Any:
+    owner_strings = ["Grid models", grid_model_name, "Grid"]
+    type_string = "Create Grid"
+    grid_job = Job.get_job(owner_strings, type_string, rms_grid_job_name)
+    return grid_job.get_arguments()
+
+
+def get_job_names(grid_model_name: str) -> list[str]:
+    owner_strings = ["Grid models", grid_model_name, "Grid"]
+    type_string = "Create Grid"
+    return Job.get_job_names(owner_strings, type_string)
+
+
+def check_and_get_grid_job_name(
+    grid_model_name: str, debug_level: int = DEBUG_OFF
+) -> Union[None, str]:
+    """Check if there are one unique grid building job for the grid model.
+    If one unique grid buidling job, check that no zone bounaries are defined by
+    the option 'Sample' and that all are defined by the method 'Honor'.
+    Return job name or None
+    """
+    job_names = get_job_names(grid_model_name)
+    if job_names is None or len(job_names) == 0:
+        if debug_level >= DEBUG_ON:
+            print("WARNING: ")
+            print(
+                f"         No automatic grid conformity check for '{grid_model_name}'"
+            )
+            print("         This grid is not created by an RMS grid building job.")
+        return None
+
+    if len(job_names) > 1:
+        if debug_level >= DEBUG_ON:
+            print("WARNING:")
+            print(
+                f"         No automatic grid conformity check for '{grid_model_name}'"
+            )
+            print("         There exists multiple grid building jobs for this grid.")
+            print("         Cannot know which one to use to check grid conformity.")
+            print("         The jobs are:")
+            print(f"        {job_names}")
+        return None
+
+    rms_grid_job_name = job_names[0]
+
+    arguments = get_grid_building_job_arguments(grid_model_name, rms_grid_job_name)
+    sample_zone_boundaries = arguments["SampledHorizons"]
+
+    err = False
+    for indx, is_sampled in enumerate(sample_zone_boundaries):
+        if is_sampled:
+            # This indicates that some zone borders are calculated using 'Sample'
+            # and not 'Honor'. In this case the conformal_mode_list no longer
+            # match the modelling zones since sone zones are merged when
+            # defining grid layout.
+            err = True
+            if debug_level >= DEBUG_ON:
+                print("WARNING: ")
+                print(
+                    f"         The grid definition in the job '{rms_grid_job_name}' "
+                    f"for the grid model '{grid_model_name}' is not implemented."
+                )
+                print(
+                    f"         Zone boundary {indx + 1} is defined by the "
+                    "option 'Sample'."
+                )
+                print(
+                    "         Only zone boundaries defined by option 'Honor' is "
+                    "implemented."
```

Comment:
Same here; use the internal function (see above)

Reply:
Fixed